extensions/amp-form/0.1/amp-form.js

```diff
@@ -690,7 +690,7 @@ export class AmpForm {
         }).then(
             resp => this.handleSsrTemplateSuccess_(resp, request),
             error => this.handleSsrTemplateFailure_(
-                /** @type {!JsonObject} */ (error)));
+                /** @type {!Error} */ (error)));
```

Comment:
Actually this might not be right. A bit confused here -- this is coming from the viewer (which also uses `postMessage` and requires cloneable objects), so how did an `Error` sneak in?

Reply:
right. fixed.

Reply:
Sorry I mean I'm not sure about the exact path of the bug. The issue appears to be an `Error` object is sent to amp-bind's worker, but where does it come from?

Reply:
Ah gotcha, The error is from the browser itself. So this was reproduced by toggling the offline radio button on the network tab of chrome dev tools. So the change I made to revert it back to jsonobject is correct as that is what's expected from the viewer but it is not coerced to a json object when the error is from an offline network.

Reply:
Ah, so the `error` object being an `Error` type wouldn't happen in practice? What's the root cause of `b/131633888` then?

Reply:
Right. According to Josh

> When a reply/resolve XHR fails, the input fields should be re-enabled (they are disabled while the XHR is pending).
> 
> See this screencast, where I make the reply XHR fail by turning off network before clicking "Reply":

So I am not sure if this ever happened in practice or whether it was just him testing the feature by toggling network. I'll ask

Reply:
Hm, I'm also not sure how toggling network causes an `Error` to be passed here. Might need more debugging.

Reply:
As discussed, it's due to viewer messaging promise rejection: https://github.com/ampproject/amphtml/blob/f6bc6fe3304aba0feec078116f8e74b0f84ea98d/extensions/amp-viewer-integration/0.1/messaging/messaging.js#L341-L354

Reply:
as discovered and discussed offline, the viewer will throw an error.